app/views/users/new.html.erb

```diff
@@ -1,3 +1,6 @@
+<!-- WE NEED TO USE THE DEVISE VIEWS FOR THE FORM -->
```

Comment:
Devise creates the `registrations/new` view, which takes care of this for you, and should be the one your header link goes to. I'd suggest double checking that, and if it's all working, delete this view.